src/full/Agda/Interaction/FindFile.hs

```diff
@@ -87,15 +88,33 @@ mkInterfaceFile fp = do
 toIFile :: SourceFile -> TCM AbsolutePath
 toIFile (SourceFile src) = do
   let fp = filePath src
-  mroot <- ifM (optLocalInterfaces <$> commandLineOptions)
-               {- then -} (pure Nothing)
-               {- else -} (libToTCM $ findProjectRoot (takeDirectory fp))
-  pure $ replaceModuleExtension ".agdai" $ case mroot of
-    Nothing   -> src
+  let localIFile = replaceModuleExtension ".agdai" src
+  mroot <- libToTCM $ findProjectRoot (takeDirectory fp)
+  case mroot of
+    Nothing   -> pure localIFile
     Just root ->
       let buildDir = root </> "_build" </> version </> "agda"
-          fileName = makeRelative root fp
-      in mkAbsolute $ buildDir </> fileName
+          fileName = makeRelative root (filePath localIFile)
+          separatedIFile = mkAbsolute $ buildDir </> fileName
+          selectIFile = ifM (optLocalInterfaces <$> commandLineOptions)
+            (pure localIFile)
+            (pure separatedIFile)
+      in do
+        separatedIFileExists <- liftIO $ doesFileExistCaseSensitive $ filePath separatedIFile
+        localIFileExists <- liftIO $ doesFileExistCaseSensitive $ filePath localIFile
+        case (separatedIFileExists, localIFileExists) of
+          (False, False) -> selectIFile
+          (False, True) -> pure localIFile
+          (True, False) -> pure separatedIFile
+          (True, True) -> do
+            ifile <- selectIFile
+            genericWarning $ P.vcat
```

Comment:
We want to get away from `GenericWarning`, because warnings should be switchable individually.
With this we get:
```
warning: -W[no]GenericWarning
There are two interface files:
...
```
So this warning can only be switched off if you also switch off all other generic warnings.

Reply:
I'll try to address this in #7112.